parsec/core/fs/storage/chunk_storage.py

```diff
@@ -289,3 +302,96 @@ def _thread_target(cursor: Cursor) -> list[ChunkID]:
 
             # Type detection is broken here for some reason
             return cast(list[ChunkID], result)
+
+    # Remanent interface
+
+    async def _load_block_remanent(self) -> None:
+        async with self._open_cursor() as cursor:
+            cursor.execute("SELECT block_remanent FROM remanence")
+            rep = cursor.fetchone()
+            self._block_remanent = rep and rep[0]
+
+    def is_block_remanent(self) -> bool:
+        return self._block_remanent
+
+    async def enable_block_remanence(self) -> bool:
+        async with self._open_cursor() as cursor:
+            # Check if remanence is already enabled
+            if self._block_remanent:
+                return False
+
+            # Use a thread as executing a statement that modifies the content of the database might,
+            # in some case, block for several hundreds of milliseconds
+            await self.localdb.run_in_thread(
+                cursor.execute,
+                """INSERT OR REPLACE INTO
+                    remanence (_id, block_remanent)
+                    VALUES (0, 1)""",
+            )
+
+            # Set remanent flag
+            self._block_remanent = True
+
+        # Indicate that the state has changed
+        return True
+
+    async def disable_block_remanence(self) -> list[ChunkID] | None:
+        async with self._open_cursor() as cursor:
+            # Check if remanence is already disabled
+            if not self._block_remanent:
+                return None
+
+            # Use a thread as executing a statement that modifies the content of the database might,
+            # in some case, block for several hundreds of milliseconds
+            await self.localdb.run_in_thread(
+                cursor.execute,
+                """INSERT OR REPLACE INTO
+                    remanence (_id, block_remanent)
+                    VALUES (0, 0)""",
+            )
+
+            # Set remanent flag
+            self._block_remanent = False
+
+            # Cleanup blocks and indicate that the state has changed
+            return await self.cleanup(cursor)
+
+    async def clear_unreferenced_chunks(
+        self, chunk_ids: list[ChunkID], not_accessed_after: float
```

Comment:
Add comment to explain `not_accessed_after`, this is pretty weird to have a timestamp as float instead of DateTime